examples/features/advancedtls/server/main.go

```diff
@@ -0,0 +1,188 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"time"
+
+	pb "google.golang.org/grpc/examples/features/proto/echo"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/credentials/tls/certprovider"
+	"google.golang.org/grpc/credentials/tls/certprovider/pemfile"
+	"google.golang.org/grpc/security/advancedtls"
+)
+
+type server struct {
+	pb.UnimplementedEchoServer
+	name string
+}
+
+const credRefreshInterval = 1 * time.Minute
+const goodServerWithCrlPort int = 50051
+const revokedServerWithCrlPort int = 50052
+const insecurePort int = 50053
+
+func (s *server) UnaryEcho(ctx context.Context, req *pb.EchoRequest) (*pb.EchoResponse, error) {
+	return &pb.EchoResponse{Message: req.Message}, nil
+}
+
+func tlsServers(credentialsDirectory string) {
+	go func() {
+		createAndRunTLSServer(credentialsDirectory, false, goodServerWithCrlPort)
+	}()
```

Comment:
`go createAndRunTLSServer(...)` instead please.  And elsewhere.

But also in Go it's an antipattern to have a func spawn something in a goroutine vs. just putting `go` on the call to the func.  Probably delete these two functions and call `createAndRun*` from `main`.  This also ties with the comment at the bottom..........

Reply:
Good call-out on the anti-pattern, fixed